src/main/groovy/org/mockito/release/notes/internal/DefaultImprovement.java

```diff
@@ -87,4 +87,40 @@ public String toJson() {
     public void toJson(Writer writable) throws IOException {
         writable.append(toJson());
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Got it. Thank you explanation!

+1 to the fix
-1 to extra complexity due to equals+hashcode (extra code to maintain, we will have to always remember to update the methods when we change/refactor how the domain objects keep state. If we forget, we are exposed to subtle bugs).

One cool way of making assertions in tests for complex domain objects is using toString(), for example [here](https://github.com/mockito/mockito-release-tools/blob/3b98d8d413c12ba9ff3ad58a8a5c0b2981088ec9/src/test/groovy/org/mockito/release/notes/vcs/DefaultReleasedVersionsProviderTest.groovy#L42). Since toString() is used mostly for debugging, it's safer than real hashcode+equals. Plus, you can tweak toString() to show meaningful stuff, in compact form. 

No action needed, let's move on ;)